service_manager/service_manager.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+from yapapi.log import enable_default_logger
+from .service_wrapper import ServiceWrapper
+from .yapapi_connector import YapapiConnector
+from functools import partial
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from typing import Type, List, Any, Callable, Awaitable
+    from yapapi.services import Service
+
+
+async def stop_on_golem_exception(service_manager: 'ServiceManager', e: Exception) -> None:
+    print("GOLEM FAILED\n", e)
+    print("STOPPING THE LOOP")
+    loop = asyncio.get_event_loop()
+    loop.stop()
+
+
+class ServiceManager():
+    def __init__(
+        self,
+        executor_cfg: dict,
+        golem_exception_handler: 'Callable[[ServiceManager, Exception], Awaitable[None]]' = stop_on_golem_exception,
```

Comment:
If the result is ignored, there's no need to restrict it to `None`.
```suggestion
        golem_exception_handler: 'Callable[[ServiceManager, Exception], Awaitable[Any]]' = stop_on_golem_exception,
```